File: 1009.c
#include <stdio.h>
int main(){
    int precalc[9][4]={
        {0,0,0,0},
        {0,0,0,0},
        {6,2,4,8},
        {1,3,9,7},
        {0,0,0,0},
        {0,0,0,0},
        {0,0,0,0},
        {1,7,9,3},
        {6,8,4,2},
    };
    int a,b;
    int t;
    scanf("%d", &t);
    for (int i=0; i<t; i++){
        scanf("%d %d",&a,&b);
        a%=10;
        if (a==0) printf("10");
        else if (a==1) printf("1");
        else if (a==4) printf(b%2 ? "4" : "6");
        else if (a==5) printf("5");
        else if (a==6) printf("6");
        else if (a==9) printf(b%2 ? "9" : "1");
        else printf("%d",precalc[a][b%4]);
        printf("\n");
    }
    return 0;
}

File: 1009.py
#깡으로 곱해주는 문제로 착각 ㄴㄴ
precalc=[0,0,[6,2,4,8],[1,3,9,7],0,0,0,[1,7,9,3],[6,8,4,2]] #2, 3, 7, 8
input=open(0).readline
for _ in range(int(input())):
    a,b=map(int,input().split())
    a%=10
    if a==0: print(10)
    elif a==1: print(1)
    elif a==4: print(4 if b%2 else 6)
    elif a==5: print(5)
    elif a==6: print(6)
    elif a==9: print(9 if b%2 else 1)
    else: print(precalc[a][b%4])